### PR TITLE
update parser

### LINKED ISSUE #6: Parser: Map validation
We can do this with either a floodfill algo or just check if all the '1' characters in the map are surrounded by '0' charaters. 
also the player character 'N/S/W/E' needs to be check in this way but with '1/0'. To make sure it isn't outside of the map.

### PR DESCRIPTION
**Pull request by Tentanus**
- [x] Proper transfer of map and data into program
- [x] Map: flood-fill
- [x] Map: validation
- [x] removed get_next_line

### CODE REVIEW

File: src/parser/parse_map.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+bool	parse_map(t_map *chart)
+{
+	int	i;
+
+	i = 0;
+	while (chart->map[i])
+	{
+		if (ft_strsubset(chart->map[i], MAP_CHARACTER))
```
Comment (Tentanus) [inline src/parser/parse_map.c:88]: Is the SEGV Issue Fixed?
@Solypsist

File: src/parser/parse_map.c [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+static bool	is_player_spawn(char point)
+{
+	if (point == 'N' || point == 'E' || point == 'S' || point == 'W')
+		return (true);
+	return (false);
+}
+
+static bool	set_player_values(t_map *chart)
+{
```
Comment (Tentanus) [inline src/parser/parse_map.c:26]: Do you check there are multiple player characters?
@Solypsist

File: src/parser/get_map.c [hunk 3]
```diff
@@ -11,30 +11,73 @@
...
+	return (ret);
+}
+
+static char	**get_map_line(char *line, ssize_t *idx)
+{
+	size_t	line_count;
+	size_t	i;
+	char	**ret;
+
+	line_count = get_line_amount(line, idx);
```
Comment (Tentanus) [inline src/parser/get_map.c]: @Tentanus 
get_line_amount doesn't work if last line doesn't end with '\n'. It's working on the final position in the array.
pl0x fix

File: src/parser/get_map.c [hunk 4]
```diff
@@ -11,30 +11,73 @@
...
+{
+	char	*ret;
+	size_t	len;
+
+	len = *idx;
+	while (line[len] && line[len++] != '\n')
+		;
+	len -= *idx;
+	ret = ft_substr(line, *idx, len);
+	*idx += len;
```
Comment (Tentanus) [inline src/parser/get_map.c]:
```suggestion
	ret = ft_substr(line, *idx, len - *idx);
	*idx = len;
```

File: src/parser/check_path.c [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	i = 0;
+	while (map[i])
+		i++;
+	copy = ft_calloc(i + 1, sizeof(char *));
+	if (copy == NULL)
+		return (cbd_error(ERR_MEMORY), NULL);
+	i = 0;
+	while (map[i])
+	{
+		copy[i] = ft_strdup(map[i]);
```
Comment (Tentanus) [inline src/parser/check_path.c:90]: what if strdup fials?
@Solypsist

**Review by Tentanus — COMMENTED** [review]
Some minor issues but it looks great so far

File: src/parser/parse_map.c (hunk 2 above)
Comment (Solypsist) [reply src/parser/parse_map.c:26]: I did not, good find. It has been fixed now.

File: src/parser/check_path.c (hunk 5 above)
Comment (Solypsist) [reply src/parser/check_path.c:90]: Then everything goes bad. I have now added a protection.

File: src/parser/get_map.c (hunk 4 above)
Comment (Solypsist) [reply src/parser/get_map.c]: This looks good.
There is a while loop in this same function that is a bit cursed:
(
while (line[len] && line[len++] != '\n')
    ;
)

File: src/parser/parse_map.c (hunk 1 above)
Comment (Solypsist) [reply src/parser/parse_map.c:88]: I cannot find a SEGV originating from this function.
If you can still replicate the SEGV issue, please let me know.

**Pull request merged**